site/src/sphinx/server-annotated-service.rst

```diff
@@ -0,0 +1,718 @@
+
+.. code-block:: java
+
+    class MyDecorator implements DecoratingServiceFunction<HttpRequest, HttpResponse> {
+        @Override
+        public HttpResponse serve(Service<HttpRequest, HttpResponse> delegate,
+                                  ServiceRequestContext ctx,
+                                  HttpRequest req) throws Exception {
+            ... Do something ...
+            return delegate.serve(ctx, req);
+        }
+    }
+
+Then, annotate your class or method with a `@Decorator`_ annotation. In the following example, ``MyDecorator``
+will handle a request first, then ``AnotherDecorator`` will handle the request next, and finally ``hello()``
+method will handle the request.
+
+.. code-block:: java
+
+    @Decorator(MyDecorator.class)
+    class MyAnnotatedServiceClass {
+
+        @Decorator(AnotherDecorator.class)
+        @Get("/hello")
+        public HttpResponse hello() { ... }
+    }
+
+Not that the applying order of the decorators is slightly different from that of the converters and exception
+handlers. As you read in :ref:`more-about-converters-exception-handlers`, both the converters and exception
+handlers are applied in the order of method-level ones, class-level ones and global ones. Unlike them,
+decorators are applied in the opposite order as follows.
+
+.. code-block:: java
+
+    @Decorator(MyClassDecorator2.class)                 // order 2
+    @Decorator(MyClassDecorator3.class)                 // order 3
+    class MyAnnotatedServiceClass {
+
+        @Get("/hello")
+        @Decorator(MyMethodDecorator4.class)            // order 4
+        @Decorator(MyMethodDecorator5.class)            // order 5
+        public HttpResponse hello() { ... }
+    }
+
+    ...
+
+    sb.annotatedService(new MyAnnotatedServiceClass(),
+                        new MyGlobalDecorator1());      // order 1
+
+
+.. _media_type_nego:
+
+Media type negotiation
+----------------------
+
+Armeria provides `@ProduceType`_ and `@ConsumeType`_ annotations to support media type negotiation. It is not
+necessary if you have only one service method for a path and an HTTP method. Assume that you have multiple
+service methods for the same path and the same HTTP method as follows.
+
+.. code-block:: java
+
+    class MyAnnotatedServiceClass {
+
+        @Get("/hello")
+        public HttpResponse hello1() {
+            // Return a text document to the client.
+        }
+
+        @Get("/hello")
+        public HttpResponse hello2() {
+            // Return a JSON object to the client.
+        }
+    }
+
+If no media type is specified on the method, the first method is usually mapped to a request for the path
+``/hello`` and the second method will be ignored so that a client would always receive a text document
+from a server. What should you do if you want to get a JSON object from the path ``/hello``? You can just
+specify the type of the content what your method produces as follows and add ``Accept`` header to your request.
```

Comment:
- what your -> your
- to your client request (to clarify that this is a client-side modification)